jlm/llvm/opt/alias-analyses/Andersen.cpp

```diff
@@ -47,13 +61,21 @@ Andersen::AnalyzeSimpleNode(const rvsdg::simple_node & node)
     AnalyzeConstantAggregateZero(node);
   else if (is<ExtractValue>(op))
     AnalyzeExtractValue(node);
-  else if (is<FreeOperation>(op))
+  else if (is<valist_op>(op))
+    AnalyzeValist(node);
+  else if (is<FreeOperation>(op) || is<ptrcmp_op>(op))
   {
-    // A free does not affect any Points-to-sets
+    // These operations take pointers as input, but do not affect any points-to sets
   }
   else
   {
-    // TODO: Make sure all simple nodes involving pointers are correctly handled by the analysis
+    // This node operation is unknown, make sure it doesn't consume any pointers
+    for (size_t i = 0; i < node.ninputs(); i++)
+    {
+      if (IsTypePointing(node.input(i)->type()))
+        std::cerr << node.operation().debug_string() << std::endl;
+      JLM_ASSERT(!IsTypePointing(node.input(i)->type()));
+    }
```

Comment:
You only check the inputs. Why not the outputs of the node?

Reply:
I just saw that you also added the same checks for the arguments and results of a region. In combination with this check, it should then cover everything. Ignore the above comment, please.